tls-tutorial/doc/use_your_own_certificate.md

```diff
@@ -0,0 +1,685 @@
+
+As always let's take a peek at the certificate that we just created.
+
+```shell
+openssl x509 -in server.crt -text -noout
+```
+
+Your server certificate should now look similar to the output below:
+
+```
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number:
+            2c:18:91:ff:3a:b3:31:38:a4:16:fc:05:f3:ed:18:74:fd:c5:57:b8
+        Signature Algorithm: sha256WithRSAEncryption
+        Issuer: CN = TLS Tutorial CA, C = DE, L = Bavaria, O = Tutorial Organization
+        Validity
+            Not Before: Jan 12 09:07:04 2023 GMT
+            Not After : Apr 12 09:07:04 2023 GMT
+        Subject: CN = TLS Tutorial MySQL Server, C = DE, L = Bavaria, O = Tutorial Organization
+        Subject Public Key Info:
+            Public Key Algorithm: rsaEncryption
+                RSA Public-Key: (4096 bit)
+                Modulus:
+                    00:99:af:[...]
+                Exponent: 65537 (0x10001)
+        X509v3 extensions:
+            X509v3 Key Usage: critical
+                Digital Signature, Non Repudiation, Key Encipherment, Key Agreement
+            X509v3 Basic Constraints: 
+                CA:FALSE
+    Signature Algorithm: sha256WithRSAEncryption
+         75:41:37:[...]
+```
+
+No we are all set to install the certificates.
+
+### Certificate Chains
+
+For the sake of completeness it should be mentioned here that out in the wild server certificates are usually signed by subordinate CAs. So the client has to follow the certificate chain all the way down to a CA it trusts. Servers can even present server certificates that are combined with some intermediate CA certificates to fill the gaps. Even then the fact still remains, that client must know and trust the last CA in the chain.
+
+We keep things simple here though, since this is just a tutorial meant to convey the general idea of server certificate validation.
+
+## Setting up MySQL
+
+1. Install MySQL client and server
+   ```shell
+   sudo apt install mysql-client mysql-server
+   ```
+2. Log into MySQL as MySQL root user (on Ubuntu now requires `sudo`)
+   ```shell
+   sudo mysql -u root
+   ```
+3. Create a database `shapes` and a user `tutorial_user` for our import experiment
+   ```sql
+   create database shapes;
+   create user 'tutorial_user' identified by 'tutorial';
+   grant all on shapes.* to tutorial_user;
+   exit
+   ```
+4. Login as a `tutorial_user`
+   ```sql
+   mysql -u tutorial_user -p
+   ```
+5. Populate the database
+   ```sql
+   create table shapes.shapes (name VARCHAR(40), corners int);
+   insert into shapes.shapes values ('point', 1), ('line', 2), ('triangle', 3), ('rectangle', 4);
+   exit
+   ```
+
+As you can tell we won't win any price for most original database usage here, but that is not the point of the tutorial.
+
+### Installing Certificates in MySQL
+
+By default, the MySQL server on Ubuntu comes with a self-signed certificate. Our next job will be to use the certificates we previously created.
```

Comment:
What’s wrong with that self-signed certificate? We should mention a good reason why we change that.

Reply:
Added explanation.